### DISCUSSION TITLE
Bazel support policy?

### DISCUSSION THREAD

**Discussion by dbolduc** [open-telemetry/opentelemetry-cpp#1886] [General]
## Background

At the time of writing, `opentelemetry-cpp` defaults to bazel version 4.2.0.
https://github.com/open-telemetry/opentelemetry-cpp/blob/9dd19a2afb254ac2bf3c55811b56628161b3c288/.bazelversion#L1

The latest bazel release is [6.0.0](https://github.com/bazelbuild/bazel/releases/tag/6.0.0). `opentelemetry-cpp` is not currently compatible with 6.0.0. e.g.
```sh
bazel build //sdk/src/common/platform:fork

# Output
ERROR: .../bazel_tools/platforms/BUILD:89:6: in alias rule @bazel_tools//platforms:windows: Constraints from @bazel_tools//platforms have been removed. Please use constraints from @platforms repository embedded in Bazel, or preferably declare dependency on https://github.com/bazelbuild/platforms. See https://github.com/bazelbuild/bazel/issues/8622 for details.
ERROR: .../bazel_tools/platforms/BUILD:89:6: Analysis of target '@bazel_tools//platforms:windows' failed
ERROR: .../opentelemetry-cpp/sdk/src/common/platform/BUILD:17:11: errors encountered resolving select() keys for //sdk/src/common/platform:fork
ERROR: Analysis of target '//sdk/src/common/platform:fork' failed; build aborted: 
```

## Google's Open Source Support Policy

`opentelemetry-cpp` [sort of depends on Abseil](https://github.com/open-telemetry/opentelemetry-cpp/blob/9dd19a2afb254ac2bf3c55811b56628161b3c288/api/BUILD#L35-L36), and the examples [depend on gRPC](https://github.com/open-telemetry/opentelemetry-cpp/blob/9dd19a2afb254ac2bf3c55811b56628161b3c288/examples/grpc/BUILD#L45).

gRPC and Abseil follow this support policy: https://opensource.google/documentation/policies/cplusplus-support#build_systems

They have dropped support for Bazel 4.x. Eventually they will require newer versions of Bazel.

## Questions

What is `opentelemetry-cpp`'s Bazel support policy?

**Comment by lalitb** [comment]
Regarding the error with Bazel v6.0.0,  PR #1873 is there for the fix. We do however need to define the policy for Bazel support, probably supporting recent two LTS versions similar to what few other does. Will discuss this in next community meeting, thanks for raising the question.

**Reply by dbolduc** [reply]
> #1873 is there for the fix

D'oh, I missed this PR. Thanks for linking.

**Reply by ncs-pl** [reply]
I think it would be better to go with 1 LTS behind only since they are actively looking to make LTS longer (the last community update proposed ~ 11 months), and with the changes incoming with Bzlmod it might be better too in the long run

**Comment by malkia** [comment]
It should work if you just enable 

```
build --incompatible_use_platforms_repo_for_constraints=false
```

I did this some time ago in my own branch (I'm using only bazel there, but it's primarirly for Windows)
https://github.com/malkia/opentelemetry-cpp/commit/3cecac5acd14068601784d0670acd0396dc2a331
